src/nbl/video/CVulkanForeignImage.h

```diff
@@ -0,0 +1,28 @@
+#ifndef __NBL_VIDEO_C_VULKAN_FOREIGN_IMAGE_H_INCLUDED__
+
+#include "nbl/video/CVulkanImage.h"
+
+namespace nbl::video
+{
+
+class CVKSwapchain;
+
+class CVulkanForeignImage final : public CVulkanImage
+{
+public:
+    CVulkanForeignImage(core::smart_refctd_ptr<ILogicalDevice>&& _vkdev,
+        IGPUImage::SCreationParams&& _params, VkImage _vkimg)
+        : CVulkanImage(std::move(_vkdev), std::move(_params), _vkimg)
+    {}
+
+    ~CVulkanForeignImage();
+
+private:
+    // circular dep
+    // core::smart_refctd_ptr<CVKSwapchain> m_swapchain; // the only foreigner we have now is swapchain
```

Comment:
actually not having any mention of swapchain in foreign image makes it more "usable"

Reply:
To be continued here #246